source/common/network/win32_socket_handle_impl.cc

```diff
@@ -18,16 +18,24 @@ namespace Network {
 
 Api::IoCallUint64Result Win32SocketHandleImpl::readv(uint64_t max_length, Buffer::RawSlice* slices,
                                                      uint64_t num_slice) {
-  auto result = IoSocketHandleImpl::readv(max_length, slices, num_slice);
-  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
-  return result;
+  if (peek_buffer_.length() == 0) {
```

Comment:
optional:  I mildly prefer other ordering here and below
if (peek_buffer_.length() !=  0) {
  return readvFromPeekBuffer(...)
}
do normal read

Reply:
will update